weave/pantry_factory.py

```diff
@@ -0,0 +1,130 @@
+from .index.index_sqlite import IndexSQLite
+from .index.index_sql import IndexSQL
+
+
+def create_pantry(**kwargs):
+    """Create a weave.Pantry object using regular params or a config file.
+
+    Three options to create a pantry:
+    1: Default args, requires Index, pantry_path, file_system
+    2: A local config file which has information about the index, path, etc
+    3: Only a pantry path and file_system, this will look for a global
+       config file in the pantry root.
+
+    Parameters:
+    -----------
+    **index: IndexABC (optional)
+        The concrete implementation of an IndexABC. This is used to track
+        the contents within the pantry.
+    **pantry_path: str (optional)
+        Path to the directory in which the pantry is located.
+    **file_system: fsspec object (optional)
+        The fsspec object which hosts the pantry we desire to index.
+        If file_system is None, then the default fs is retrieved from the
+        config.
+    **config_file: str (optional)
+    Optional kwargs passed to the Index constructor.
+
+    Returns:
+    -----------
+    weave.Pantry object
+    """
+    pantry = None
+
+    # Create a pantry in the default way using the Pantry constructor
+    if all(k in kwargs for k in ["index", "pantry_path", "file_system"]):
+        pantry = Pantry(
+            kwargs.pop("index"),
+            pantry_path=kwargs.pop("pantry_path"),
+            file_system=kwargs.pop("file_system"),
+            **kwargs,
+        )
+    # Create a pantry using a config file that is given to the pantry factory.
+    elif "config_file" in kwargs:
+        with open(
+            kwargs.pop("config_file"),
+            "r",
+            encoding="utf-8"
+        ) as config_file:
+            config = json.load(config_file)
+
+        pantry = _create_pantry_from_config(config, **kwargs)
+    # Create a pantry using a config that exists inside the pantry directory.
+    elif "pantry_path" in kwargs:
+        file_system = kwargs.pop("file_system")
+        if file_system is None:
+            file_system = get_file_system()
+
+        with file_system.open(
+            os.path.join(kwargs.pop("pantry_path"), "config.json"),
+            'r'
+        ) as config_file:
+            config = json.load(config_file)
+
+        pantry = _create_pantry_from_config(config, **kwargs)
+    # If pantry was unable to be created using the methods above, throw error.
+    else:
+        raise ValueError("Invalid kwargs passed, unable to make pantry")
+    return pantry
+
+
+def _create_pantry_from_config(config, **kwargs):
+    """Create the weave.pantry object from a pre-existing config file.
+
+    Parameters:
+    -----------
+    config: dict
+        Dictionary that contains the contents of the config.json file.
+    Optional kwargs passed to the Index constructor.
+
+    Returns:
+    -----------
+    weave.Pantry object
+    """
+    # Parse the index type used by this pantry.
+    index_type = config["index"]
```

Comment:
Should you catch if there is no 'index' in the config file? Same with 'pantry_path' and 'file_system' or are we assuming the user will make their config correctly?